File: frontend/src/types/report.ts
export interface StatusCount {
  status: string;
  count: number;
}

export interface CategoryStat {
  category: string;
  sales: number;
  orders: number;
}

export interface CustomerStat {
  name: string;
  orders: number;
  spending: number;
}

export interface ReportSummary {
  current_revenue: number;
  previous_revenue: number;
  growth_pct: number;
  status_breakdown: StatusCount[];
  category_breakdown: CategoryStat[];
  top_customers: CustomerStat[];
}

File: backend/app/schemas/report.py
from pydantic import BaseModel


class StatusCount(BaseModel):
    status: str
    count: int


class CategoryStat(BaseModel):
    category: str
    sales: float
    orders: int


class CustomerStat(BaseModel):
    name: str
    orders: int
    spending: float


class ReportSummary(BaseModel):
    current_revenue: float
    previous_revenue: float
    growth_pct: float
    status_breakdown: list[StatusCount]
    category_breakdown: list[CategoryStat]
    top_customers: list[CustomerStat]
